### PR TITLE
feat: addresses whitelist (DAC-485)

### PR DESCRIPTION
**Pull request by bogdan-manole**
add support for whitelisted addresses access only (this is an optional run mode of the service)

### CODE REVIEW

File: server/Main.hs [hunk 1]
```diff
@@ -335,13 +354,16 @@ main = do
                      { corsRequestHeaders = ["Content-Type", "Authorization"]
                      , corsMethods = [methodGet,methodPost,methodPut,methodDelete,methodHead]
                      }
+      -- get the whitelisted addresses from $WLIST evn var
+      -- if useWhitelist is set to true, then the whitelist is ignored
```
Comment (RSoulatIOHK) [inline server/Main.hs:358]: Shouldn't the comment be "if useWhitelist is set to false, the the whitelist is ignored?"

File: server/Main.hs [hunk 2]
```diff
@@ -335,13 +354,16 @@ main = do
                      { corsRequestHeaders = ["Content-Type", "Authorization"]
                      , corsMethods = [methodGet,methodPost,methodPut,methodDelete,methodHead]
                      }
+      -- get the whitelisted addresses from $WLIST evn var
```
Comment (RSoulatIOHK) [inline server/Main.hs:357]: "env var"?

**Review by RSoulatIOHK — CHANGES_REQUESTED** [review]
The addition of the whitelist looks ok. Just a few remarks on some comments that seemed erroneous.

File: server/Main.hs [hunk 3]
```diff
@@ -269,22 +280,30 @@ jwtAuthHandler secret = mkAuthHandler handler
...
+          -- verify if the address is whitelisted
+          verifyWhiteList whitelist addr
           -- compare the expiration time with the current time
           when (now > expiration) $ throw401 "JWT token expired"
           pure (toId pid, UserAddress addr)
     Nothing -> throw401 "Missing Authorization header"
   extractToken =
     let isSpace = (== ' ')
     in BS.dropWhileEnd isSpace . BS.dropWhile isSpace . snd . BS.break isSpace
+  -- get the whitelisted addresses from $WLIST evn var
```
[the remark this answers is not in the document]
Comment (bogdan-manole) [reply server/Main.hs]: done, thanks

File: server/Main.hs (hunk 2 above)
Comment (bogdan-manole) [reply server/Main.hs]: done

File: server/Main.hs (hunk 1 above)
Comment (bogdan-manole) [reply server/Main.hs]: done

**Review by RSoulatIOHK — APPROVED** [review]

**Pull request merged**